code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -199,9 +209,12 @@
 			var/mob/living/carbon/monkey/food = new /mob/living/carbon/monkey(remote_eye.loc, TRUE, owner)
 			if (!QDELETED(food))
 				food.LAssailant = C
-				X.monkeys --
-				to_chat(owner, "[X] now has [X.monkeys] monkeys left.")
-	else
+				X.monkeys--
+				X.monkeys = FLOOR(X.monkeys, 0.1)		//Prevents rounding errors
```

Comment:
Where's the crazy math that requires this safeguard or perhaps what's the reason behind it?

If you're only working with tenths it seems pretty silly to "make sure" you're working with tenths.

Reply:
Floats is the reason. It would cause chat outputs like: "[console] now has 0.60000032  monkeys available. "
I suppose it should strictly speaking be using rounding instead of flooring, since it can go both ways. Other option is formatting the chat output, but I rather like neat numbers in my vars.

I'll replace the floors with round instead though.

Reply:
I'm asking what math are you doing to get the 0.6**000000032**

Reply:
Unlike integer types, floats are inaccurate and often cause issues like this. This is not DM specific, it is just a shortcoming of floating point numbers in code.

This site explains it well if you are interested: [link](http://effbot.org/pyfaq/why-are-floating-point-calculations-so-inaccurate.htm)

Reply:
Floats are inaccurate at high values. Have you actually TRIED and saw that it doesn't truncate, or are you just applying "good coding practices" at random?

Reply:
@Barhandar This was a fix for that specific issue, as you can see in the commit history. Floats are inaccurate at all values.

@ExcessiveUseOfCobblestone You can't represent 0.8 in integers. Does DM even have integers? I thought I read it was floats only, but I might be mixing it up with something else.